maze_transformer/training/training.py

```diff
@@ -88,13 +94,12 @@ def train(
         cfg.train_cfg.checkpoint_interval // cfg.train_cfg.batch_size
     )
     for iteration, batch in enumerate(dataloader):
-        # compute loss
-        # batch_on_device: Int[torch.Tensor, "batch sequence"] = batch.type(
-        #     dtype=torch.LongTensor
-        # ).to(model.cfg.device)
-
-        # loss: Loss = model(batch_on_device[:, :-1], return_type="loss")
-        loss: Loss = model(batch, return_type="loss")
+        loss: Float[torch.Tensor, ""]
```

Comment:
I don't thinks `loss` is actually a plain tensor here, I'm pretty sure its a `TransformerLens` Loss object?

Reply:
Good point! I think we can narrow it down though, in this case we know it is actually a `SingleLoss` because we're not setting `loss_per_token=True` in our forward call. 

These are the relevant TL types:
```
SingleLoss = Float[torch.Tensor, ""] # Type alias for a single element tensor
LossPerToken = Float[torch.Tensor, "batch pos-1"]
Loss = Union[SingleLoss, LossPerToken]
```

Definitely better to use `SingleLoss` here rather than what I've got, I'll make that change. Also open to changing it back to `Loss` if you think that's best.